server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1583,12 +1583,27 @@ public Volume destroyVolume(long volumeId, Account caller, boolean expunge, bool
                 s_logger.warn("Failed to expunge volume: " + volumeId);
                 return null;
             }
+            markVolumeSnapshotsAsDestroyed(volume);
```

Comment:
@weizhouapache, this is not applied for the ROOT volumes. If the `snapshot.backup.to.secondary` is disabled and you have snapshots on the ROOT volume when you expunge the VM the snapshots will be in a `BackedUp` state in the DB table `snapshots`, but with no records in the `snapshot_store_ref` table, it is removed.